src/librustc/hir/check_attr.rs

```diff
@@ -35,11 +35,26 @@ impl Target {
     }
 }
 
+/// Visitor which check if all attributes are valid.
+///
+/// Note: Doesn't check if attributes are existing.
```

Comment:
I'm not sure if this doc comment is useful. I would prefer a module-level comment saying something like:
```rust
//! This module implements some validity checks for attributes.
//! In particular it verifies that `#[inline]` and `#[repr]` attributes are 
//! attached to items that actually support them and if there are
//! conflicts between multiple such attributes attached to the same 
//! item.
```